packages/clerk-js/src/core/modules/debug/logger.ts

```diff
@@ -0,0 +1,185 @@
+
+/**
+ * Default log level for debug logging
+ */
+const DEFAULT_LOG_LEVEL: DebugLogLevel = 'debug';
+
+/**
+ * Minimal debug logger interface for engineers.
+ *
+ * @public
+ */
+export class DebugLogger {
+  private readonly filters?: DebugLogFilter[];
+  private readonly logLevel: DebugLogLevel;
+  private readonly transport: DebugTransport;
+
+  /**
+   * Creates a new debug logger.
+   *
+   * @param transport - Transport used to send log entries
+   * @param logLevel - Minimum log level to record. Defaults to `'debug'`
+   * @param filters - Optional list of filters to include or exclude messages
+   */
+  constructor(transport: DebugTransport, logLevel?: DebugLogLevel, filters?: DebugLogFilter[]) {
+    this.transport = transport;
+    this.logLevel = logLevel ?? DEFAULT_LOG_LEVEL;
+    this.filters = filters;
+  }
+
+  /**
+   * Log a debug message.
+   *
+   * @param message - Text description of the event
+   * @param context - Optional structured context to attach
+   * @param source - Optional logical source identifier
+   */
+  debug(message: string, context?: Record<string, unknown>, source?: string): void {
+    this.log('debug', message, context, source);
+  }
+
+  /**
+   * Log an error message.
+   *
+   * @param message - Text description of the event
+   * @param context - Optional structured context to attach
+   * @param source - Optional logical source identifier
+   */
+  error(message: string, context?: Record<string, unknown>, source?: string): void {
+    this.log('error', message, context, source);
+  }
+
+  /**
+   * Log an informational message.
+   *
+   * @param message - Text description of the event
+   * @param context - Optional structured context to attach
+   * @param source - Optional logical source identifier
+   */
+  info(message: string, context?: Record<string, unknown>, source?: string): void {
+    this.log('info', message, context, source);
+  }
+
+  /**
+   * Log a trace message.
+   *
+   * @param message - Text description of the event
+   * @param context - Optional structured context to attach
+   * @param source - Optional logical source identifier
+   */
+  trace(message: string, context?: Record<string, unknown>, source?: string): void {
+    this.log('trace', message, context, source);
+  }
+
+  /**
+   * Log a warning message.
+   *
+   * @param message - Text description of the event
+   * @param context - Optional structured context to attach
+   * @param source - Optional logical source identifier
+   */
+  warn(message: string, context?: Record<string, unknown>, source?: string): void {
+    this.log('warn', message, context, source);
+  }
+
+  private log(level: DebugLogLevel, message: string, context?: Record<string, unknown>, source?: string): void {
+    if (!this.shouldLogLevel(level)) {
+      return;
+    }
+
+    if (!this.shouldLogFilters(level, message, source)) {
+      return;
+    }
+
+    const entry: DebugLogEntry = {
+      id: generateUuid(),
```

Comment:
consider moving the ID generation to the API, I'm not sure we need it here, and it avoids uuid generation complexity on the client if we drop it